rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -45,31 +46,38 @@ SqliteStorage::SqliteStorage()
 {}
 
 void SqliteStorage::open(
-  const std::string & uri, rosbag2_storage::storage_interfaces::IOFlag io_flag)
+  const std::string & uri, rosbag2_storage::storage_interfaces::IOFlag io_flag,
+  const uint64_t max_bagfile_size)
 {
   auto metadata = is_read_only(io_flag) ?
     load_metadata(uri) :
     std::unique_ptr<rosbag2_storage::BagMetadata>();
 
+  std::string database_name;
+
   if (metadata) {
     if (metadata->relative_file_paths.empty()) {
       throw std::runtime_error(
               "Failed to read from bag '" + uri + "': Missing database file path in metadata");
     }
 
-    database_name_ = metadata->relative_file_paths[0];
+    database_name = metadata->relative_file_paths[0];
   } else {
     if (is_read_only(io_flag)) {
       throw std::runtime_error("Failed to read from bag '" + uri + "': No metadata found.");
     }
 
-    database_name_ = rosbag2_storage::FilesystemHelper::get_folder_name(uri) + ".db3";
+    std::stringstream ss;
+    ss << rosbag2_storage::FilesystemHelper::get_folder_name(uri) << "_" <<
+      database_file_counter_ << ".db3";
+    database_name = ss.str();
   }
+  database_names_.push_back(database_name);
```

Comment:
with `database_name_` being a vector right now, I feel this `push_back` should come after the error checking if the database_path exists.
The way it is right now would mean that if even though the exception in line 80 would be thrown, the non-existing database path would be still part of the vector.